ckan/model/user.py

```diff
@@ -87,7 +88,7 @@ class User(core.StatefulObjectMixin,
     DOUBLE_SLASH = re.compile(r':\/([^/])')
 
     @classmethod
-    def by_email(cls: Type[TUser], email: str) -> Optional["User"]:
+    def by_email(cls: Type[TUser], email: Optional[str]) -> Optional["User"]:
```

Comment:
This seems dangerous, is there a reason why we need it optional? Perhaps an `if email` check is needed further up the call stack

Reply:
my mistake sry